### ISSUE TITLE
LogStash exiting with Code 1

### ISSUE THREAD

**Issue by rohan04** [deviantony/docker-elk#79]
Hi , I am facing issues starting up the logstash. I have checked a similar issue that someone else asked and tried that solution as well, still no luck. The logstash container starts and shuts down within 2 minutes with the below error; Also I do not see logstash, elasticsearch and kibana in /etc and /var directories. Can you please help.

logstash_1       | Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
dockerelk_logstash_1 exited with code 1

Below is the docker-compose.yml

```
version: '2'

services:
  elasticsearch:
    build: elasticsearch/
    ports:
      - "9200:9200"
      - "9300:9300"
    environment:
      ES_JAVA_OPTS: "-Xms1g -Xmx1g"
    networks:
      - docker_elk
  logstash:
    build: logstash/
    command: -f /etc/logstash/conf.d/
    volumes:
      - ./logstash/config:/etc/logstash/conf.d
    ports:
      - "5000:5000"
    networks:
      - docker_elk
    environment:
      - LS_HEAP_SIZE=1g
  kibana:
    build: kibana/
    volumes:
      - ./kibana/config/:/opt/kibana/config/
    ports:
      - "5601:5601"
    networks:
      - docker_elk

networks:
  docker_elk:
    driver: bridge
```

And logstash.conf

```
input {
  tcp {
    port => 5000
    data_timeout => 600
  }
}
filter {
    mutate {
      strip => "message"
    }
    grok {   ..........
........................
   }
 
    kv {
      field_split => "&"
      source => payload
    }
    mutate {
      convert => ["T", "float"]
            remove_field =>  ["garbage", "dt_year", "dt_month",  "dt_day", "dt_hour", "dt_minute", "dt_second", "payload", "message", "CALTYPE", "FREE_MEM"]
    }
}
output {
        elasticsearch {
                host => localhost
                port => 9200
                index => "logstash-%{+YYYY.MM.dd}"
        }
} 
```

Kernel Info:
3.13.0-46-generic #77-Ubuntu

Docker Engine:

```
 docker-engine:
  Installed: 1.12.3-0~trusty
  Candidate: 1.12.3-0~trusty
  Version table:
 *** 1.12.3-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
        100 /var/lib/dpkg/status
     1.12.2-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.12.1-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.12.0-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.11.2-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.11.1-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.11.0-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.10.3-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
     1.10.2-0~trusty 0
        500 https://apt.dockerproject.org/repo/ ubuntu-trusty/main amd64 Packages
```

**Comment by deviantony** [comment]
Could you copy the logs of `docker-compose up` here too please?

**Comment by rohan04** [comment]
```
Attaching to dockerelk_logstash_1, dockerelk_elasticsearch_1, dockerelk_kibana_1
logstash_1       | Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
elasticsearch_1  | [2016-11-22T22:55:12,613][INFO ][o.e.n.Node               ] [] initializing ...
elasticsearch_1  | [2016-11-22T22:55:13,054][INFO ][o.e.e.NodeEnvironment    ] [uZAk4ml] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/disk/by-uuid/21813bed-571e-4fcc-905d-32bc13641493)]], net usable_space [14.5gb], net total_space [29.4gb], spins? [possibly], types [ext4]
elasticsearch_1  | [2016-11-22T22:55:13,059][INFO ][o.e.e.NodeEnvironment    ] [uZAk4ml] heap size [1007.3mb], compressed ordinary object pointers [true]
elasticsearch_1  | [2016-11-22T22:55:13,067][INFO ][o.e.n.Node               ] [uZAk4ml] node name [uZAk4ml] derived from node ID; set [node.name] to override
elasticsearch_1  | [2016-11-22T22:55:13,076][INFO ][o.e.n.Node               ] [uZAk4ml] version[5.0.0], pid[1], build[253032b/2016-10-26T05:11:34.737Z], OS[Linux/3.13.0-46-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_111/25.111-b14]
elasticsearch_1  | [2016-11-22T22:55:20,846][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [aggs-matrix-stats]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [ingest-common]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-expression]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-groovy]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-mustache]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-painless]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [percolator]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [reindex]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [transport-netty3]
elasticsearch_1  | [2016-11-22T22:55:20,851][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [transport-netty4]
elasticsearch_1  | [2016-11-22T22:55:20,859][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded plugin [x-pack]
elasticsearch_1  | [2016-11-22T22:55:23,386][WARN ][o.e.d.s.g.GroovyScriptEngineService] [groovy] scripts are deprecated, use [painless] scripts instead
elasticsearch_1  | [2016-11-22T22:55:33,357][INFO ][o.e.n.Node               ] [uZAk4ml] initialized
elasticsearch_1  | [2016-11-22T22:55:33,357][INFO ][o.e.n.Node               ] [uZAk4ml] starting ...
elasticsearch_1  | [2016-11-22T22:55:34,542][INFO ][o.e.t.TransportService   ] [uZAk4ml] publish_address {172.18.0.4:9300}, bound_addresses {[::]:9300}
elasticsearch_1  | [2016-11-22T22:55:34,565][INFO ][o.e.b.BootstrapCheck     ] [uZAk4ml] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
elasticsearch_1  | ERROR: bootstrap checks failed
elasticsearch_1  | max virtual memory areas vm.max_map_count [65530] likely too low, increase to at least [262144]
elasticsearch_1  | [2016-11-22T22:55:34,600][INFO ][o.e.n.Node               ] [uZAk4ml] stopping ...
elasticsearch_1  | [2016-11-22T22:55:34,741][INFO ][o.e.n.Node               ] [uZAk4ml] stopped
elasticsearch_1  | [2016-11-22T22:55:34,741][INFO ][o.e.n.Node               ] [uZAk4ml] closing ...
elasticsearch_1  | [2016-11-22T22:55:34,810][INFO ][o.e.n.Node               ] [uZAk4ml] closed
elasticsearch_1  | [2016-11-22T22:58:27,371][INFO ][o.e.n.Node               ] [] initializing ...
elasticsearch_1  | [2016-11-22T22:58:27,569][INFO ][o.e.e.NodeEnvironment    ] [uZAk4ml] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/disk/by-uuid/21813bed-571e-4fcc-905d-32bc13641493)]], net usable_space [14.6gb], net total_space [29.4gb], spins? [possibly], types [ext4]
elasticsearch_1  | [2016-11-22T22:58:27,570][INFO ][o.e.e.NodeEnvironment    ] [uZAk4ml] heap size [1007.3mb], compressed ordinary object pointers [true]
elasticsearch_1  | [2016-11-22T22:58:27,574][INFO ][o.e.n.Node               ] [uZAk4ml] node name [uZAk4ml] derived from node ID; set [node.name] to override
elasticsearch_1  | [2016-11-22T22:58:27,580][INFO ][o.e.n.Node               ] [uZAk4ml] version[5.0.0], pid[1], build[253032b/2016-10-26T05:11:34.737Z], OS[Linux/3.13.0-46-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_111/25.111-b14]
elasticsearch_1  | [2016-11-22T22:58:34,022][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [aggs-matrix-stats]
elasticsearch_1  | [2016-11-22T22:58:34,023][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [ingest-common]
elasticsearch_1  | [2016-11-22T22:58:34,023][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-expression]
elasticsearch_1  | [2016-11-22T22:58:34,024][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-groovy]
elasticsearch_1  | [2016-11-22T22:58:34,024][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-mustache]
elasticsearch_1  | [2016-11-22T22:58:34,024][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [lang-painless]
elasticsearch_1  | [2016-11-22T22:58:34,025][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [percolator]
elasticsearch_1  | [2016-11-22T22:58:34,025][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [reindex]
elasticsearch_1  | [2016-11-22T22:58:34,025][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [transport-netty3]
elasticsearch_1  | [2016-11-22T22:58:34,026][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded module [transport-netty4]
elasticsearch_1  | [2016-11-22T22:58:34,031][INFO ][o.e.p.PluginsService     ] [uZAk4ml] loaded plugin [x-pack]
elasticsearch_1  | [2016-11-22T22:58:35,393][WARN ][o.e.d.s.g.GroovyScriptEngineService] [groovy] scripts are deprecated, use [painless] scripts instead
kibana_1         | Stalling for Elasticsearch
kibana_1         | Stalling for Elasticsearch
elasticsearch_1  | [2016-11-22T22:58:46,030][INFO ][o.e.n.Node               ] [uZAk4ml] initialized
elasticsearch_1  | [2016-11-22T22:58:46,039][INFO ][o.e.n.Node               ] [uZAk4ml] starting ...
elasticsearch_1  | [2016-11-22T22:58:46,798][INFO ][o.e.t.TransportService   ] [uZAk4ml] publish_address {172.18.0.2:9300}, bound_addresses {[::]:9300}
elasticsearch_1  | [2016-11-22T22:58:46,814][INFO ][o.e.b.BootstrapCheck     ] [uZAk4ml] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
elasticsearch_1  | [2016-11-22T22:58:50,270][INFO ][o.e.c.s.ClusterService   ] [uZAk4ml] new_master {uZAk4ml}{uZAk4ml8T5u38pDL9i92ug}{V-jrBUgOTK6x7cO_ut4APw}{172.18.0.2}{172.18.0.2:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
elasticsearch_1  | [2016-11-22T22:58:50,375][INFO ][o.e.h.HttpServer         ] [uZAk4ml] publish_address {172.18.0.2:9200}, bound_addresses {[::]:9200}
elasticsearch_1  | [2016-11-22T22:58:50,375][INFO ][o.e.n.Node               ] [uZAk4ml] started
kibana_1         | Starting Kibana
elasticsearch_1  | [2016-11-22T22:58:50,766][INFO ][o.e.g.GatewayService     ] [uZAk4ml] recovered [0] indices into cluster_state
elasticsearch_1  | [2016-11-22T22:58:53,357][INFO ][o.e.m.j.JvmGcMonitorService] [uZAk4ml] [gc][7] overhead, spent [311ms] collecting in the last [1s]
elasticsearch_1  | [2016-11-22T22:58:56,513][INFO ][o.e.x.m.e.Exporters      ] [uZAk4ml] skipping exporter [default_local] as it isn't ready yet
elasticsearch_1  | [2016-11-22T22:58:56,517][ERROR][o.e.x.m.AgentService     ] [uZAk4ml] exception when exporting documents
elasticsearch_1  | org.elasticsearch.xpack.monitoring.exporter.ExportException: exporters are either not ready or faulty
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.export(Exporters.java:188) ~[x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.AgentService$ExportingWorker.run(AgentService.java:208) [x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:kibana@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["reporting","warning"],"pid":1,"message":"Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set xpack.reporting.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:06Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | [2016-11-22T22:59:06,789][INFO ][o.e.c.m.MetaDataCreateIndexService] [uZAk4ml] [.monitoring-es-2-2016.11.22] creating index, cause [auto(bulk api)], templates [.monitoring-es-2], shards [1]/[1], mappings [_default_, node, shards, index_stats, index_recovery, cluster_state, cluster_stats, indices_stats, node_stats]
elasticsearch_1  | [2016-11-22T22:59:07,175][INFO ][o.e.c.m.MetaDataCreateIndexService] [uZAk4ml] [.monitoring-data-2] creating index, cause [auto(bulk api)], templates [.monitoring-data-2], shards [1]/[1], mappings [node, kibana, cluster_info]
elasticsearch_1  | [2016-11-22T22:59:07,903][INFO ][o.e.c.m.MetaDataMappingService] [uZAk4ml] [.monitoring-es-2-2016.11.22/IMC9saa4RWWd0YZ3M6LiNQ] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-22T22:59:08,059][INFO ][o.e.c.m.MetaDataMappingService] [uZAk4ml] [.monitoring-es-2-2016.11.22/IMC9saa4RWWd0YZ3M6LiNQ] update_mapping [node_stats]
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:12Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:12Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:12Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:12Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
elasticsearch_1  | [2016-11-22T22:59:12,887][WARN ][o.e.d.i.m.StringFieldMapper$TypeParser] The [string] field is deprecated, please use [text] or [keyword] instead on [buildNum]
elasticsearch_1  | [2016-11-22T22:59:12,891][INFO ][o.e.c.m.MetaDataCreateIndexService] [uZAk4ml] [.kibana] creating index, cause [api], templates [], shards [1]/[1], mappings [server, config]
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Kibana index ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["license","info","xpack"],"pid":1,"message":"Imported license information from Elasticsearch: mode: trial | status: active | expiry date: 2016-12-22T22:58:56+00:00"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from green to yellow - Waiting for Monitoring Health Check","prevState":"green","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:13Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:16Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"Waiting for Monitoring Health Check"}
elasticsearch_1  | [2016-11-22T22:59:18,384][INFO ][o.e.c.m.MetaDataMappingService] [uZAk4ml] [.monitoring-es-2-2016.11.22/IMC9saa4RWWd0YZ3M6LiNQ] update_mapping [index_stats]
elasticsearch_1  | [2016-11-22T22:59:18,538][INFO ][o.e.c.m.MetaDataMappingService] [uZAk4ml] [.monitoring-es-2-2016.11.22/IMC9saa4RWWd0YZ3M6LiNQ] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-22T22:59:18,602][INFO ][o.e.c.m.MetaDataMappingService] [uZAk4ml] [.monitoring-es-2-2016.11.22/IMC9saa4RWWd0YZ3M6LiNQ] update_mapping [indices_stats]
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:19Z","tags":["status","plugin:security@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:19Z","tags":["security","warning"],"pid":1,"message":"Generating a random key for xpack.security.encryptionKey. To prevent sessions from being invalidated on restart, please set xpack.security.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:19Z","tags":["security","warning"],"pid":1,"message":"Session cookies will be transmitted over insecure connections. This is not recommended."}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:20Z","tags":["status","plugin:console@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:20Z","tags":["status","plugin:timelion@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:20Z","tags":["listening","info"],"pid":1,"message":"Server running at http://0.0.0.0:5601"}
kibana_1         | {"type":"log","@timestamp":"2016-11-22T22:59:20Z","tags":["status","ui settings","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | [2016-11-22T22:59:23,622][INFO ][o.e.c.m.MetaDataCreateIndexService] [uZAk4ml] [.monitoring-kibana-2-2016.11.22] creating index, cause [auto(bulk api)], templates [.monitoring-kibana-2], shards [1]/[1], mappings [_default_, kibana_stats]
logstash_1       | Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
dockerelk_logstash_1 exited with code 1

```

**Comment by deviantony** [comment]
Looks like you forgot one part in the setup:

```
elasticsearch_1  | ERROR: bootstrap checks failed
elasticsearch_1  | max virtual memory areas vm.max_map_count [65530] likely too low, increase to at least [262144]
```

Please read the documentation on how to start the stack: https://github.com/deviantony/docker-elk#increase-max_map_count-on-your-host-linux

**Comment by rohan04** [comment]
Hi @deviantony , I did try that step before starting the containers, not sure why that error occured.

Anyway , I executed that command again and restarted the containers, still the same error;

```
Attaching to dockerelk_elasticsearch_1, dockerelk_logstash_1, dockerelk_kibana_1
kibana_1         | Stalling for Elasticsearch
elasticsearch_1  | [2016-11-23T00:25:40,123][INFO ][o.e.n.Node               ] [] initializing ...
elasticsearch_1  | [2016-11-23T00:25:40,311][INFO ][o.e.e.NodeEnvironment    ] [NggTRJz] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/disk/by-uuid/21813bed-571e-4fcc-905d-32bc13641493)]], net usable_space [12.7gb], net total_space [29.4gb], spins? [possibly], types [ext4]
elasticsearch_1  | [2016-11-23T00:25:40,313][INFO ][o.e.e.NodeEnvironment    ] [NggTRJz] heap size [1007.3mb], compressed ordinary object pointers [true]
elasticsearch_1  | [2016-11-23T00:25:40,318][INFO ][o.e.n.Node               ] [NggTRJz] node name [NggTRJz] derived from node ID; set [node.name] to override
elasticsearch_1  | [2016-11-23T00:25:40,323][INFO ][o.e.n.Node               ] [NggTRJz] version[5.0.0], pid[1], build[253032b/2016-10-26T05:11:34.737Z], OS[Linux/3.13.0-46-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_111/25.111-b14]
elasticsearch_1  | [2016-11-23T00:25:46,035][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [aggs-matrix-stats]
elasticsearch_1  | [2016-11-23T00:25:46,035][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [ingest-common]
elasticsearch_1  | [2016-11-23T00:25:46,035][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [lang-expression]
elasticsearch_1  | [2016-11-23T00:25:46,035][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [lang-groovy]
elasticsearch_1  | [2016-11-23T00:25:46,036][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [lang-mustache]
elasticsearch_1  | [2016-11-23T00:25:46,036][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [lang-painless]
elasticsearch_1  | [2016-11-23T00:25:46,036][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [percolator]
elasticsearch_1  | [2016-11-23T00:25:46,036][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [reindex]
elasticsearch_1  | [2016-11-23T00:25:46,036][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [transport-netty3]
elasticsearch_1  | [2016-11-23T00:25:46,037][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded module [transport-netty4]
elasticsearch_1  | [2016-11-23T00:25:46,047][INFO ][o.e.p.PluginsService     ] [NggTRJz] loaded plugin [x-pack]
elasticsearch_1  | [2016-11-23T00:25:47,319][WARN ][o.e.d.s.g.GroovyScriptEngineService] [groovy] scripts are deprecated, use [painless] scripts instead
elasticsearch_1  | [2016-11-23T00:25:56,612][INFO ][o.e.n.Node               ] [NggTRJz] initialized
elasticsearch_1  | [2016-11-23T00:25:56,612][INFO ][o.e.n.Node               ] [NggTRJz] starting ...
elasticsearch_1  | [2016-11-23T00:25:57,508][INFO ][o.e.t.TransportService   ] [NggTRJz] publish_address {172.18.0.4:9300}, bound_addresses {[::]:9300}
elasticsearch_1  | [2016-11-23T00:25:57,520][INFO ][o.e.b.BootstrapCheck     ] [NggTRJz] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
elasticsearch_1  | [2016-11-23T00:26:01,065][INFO ][o.e.c.s.ClusterService   ] [NggTRJz] new_master {NggTRJz}{NggTRJzwT4eGoUsotkGM0A}{K5vYMzO7SUG-0HZVOqQTWw}{172.18.0.4}{172.18.0.4:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
elasticsearch_1  | [2016-11-23T00:26:01,226][INFO ][o.e.h.HttpServer         ] [NggTRJz] publish_address {172.18.0.4:9200}, bound_addresses {[::]:9200}
elasticsearch_1  | [2016-11-23T00:26:01,226][INFO ][o.e.n.Node               ] [NggTRJz] started
kibana_1         | Starting Kibana
elasticsearch_1  | [2016-11-23T00:26:01,633][INFO ][o.e.g.GatewayService     ] [NggTRJz] recovered [0] indices into cluster_state
elasticsearch_1  | [2016-11-23T00:26:04,968][INFO ][o.e.m.j.JvmGcMonitorService] [NggTRJz] [gc][8] overhead, spent [255ms] collecting in the last [1s]
elasticsearch_1  | [2016-11-23T00:26:07,084][INFO ][o.e.x.m.e.Exporters      ] [NggTRJz] skipping exporter [default_local] as it isn't ready yet
elasticsearch_1  | [2016-11-23T00:26:07,084][ERROR][o.e.x.m.AgentService     ] [NggTRJz] exception when exporting documents
elasticsearch_1  | org.elasticsearch.xpack.monitoring.exporter.ExportException: exporters are either not ready or faulty
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.export(Exporters.java:188) ~[x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.AgentService$ExportingWorker.run(AgentService.java:208) [x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:13Z","tags":["status","plugin:kibana@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:13Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:14Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:14Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:14Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:14Z","tags":["reporting","warning"],"pid":1,"message":"Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set xpack.reporting.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:14Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
logstash_1       | Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
elasticsearch_1  | [2016-11-23T00:26:17,549][INFO ][o.e.c.m.MetaDataCreateIndexService] [NggTRJz] [.monitoring-es-2-2016.11.23] creating index, cause [auto(bulk api)], templates [.monitoring-es-2], shards [1]/[1], mappings [shards, _default_, node, index_stats, index_recovery, cluster_state, cluster_stats, indices_stats, node_stats]
elasticsearch_1  | [2016-11-23T00:26:18,073][INFO ][o.e.c.m.MetaDataCreateIndexService] [NggTRJz] [.monitoring-data-2] creating index, cause [auto(bulk api)], templates [.monitoring-data-2], shards [1]/[1], mappings [node, cluster_info, kibana]
elasticsearch_1  | [2016-11-23T00:26:19,016][INFO ][o.e.c.m.MetaDataMappingService] [NggTRJz] [.monitoring-es-2-2016.11.23/-AEPo8pmRoW0eDALvM_45g] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-23T00:26:19,506][INFO ][o.e.c.m.MetaDataMappingService] [NggTRJz] [.monitoring-es-2-2016.11.23/-AEPo8pmRoW0eDALvM_45g] update_mapping [node_stats]
dockerelk_logstash_1 exited with code 1
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:20Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:20Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:20Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:20Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
elasticsearch_1  | [2016-11-23T00:26:20,480][WARN ][o.e.d.i.m.StringFieldMapper$TypeParser] The [string] field is deprecated, please use [text] or [keyword] instead on [buildNum]
elasticsearch_1  | [2016-11-23T00:26:20,481][INFO ][o.e.c.m.MetaDataCreateIndexService] [NggTRJz] [.kibana] creating index, cause [api], templates [], shards [1]/[1], mappings [server, config]
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:20Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Kibana index ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:21Z","tags":["license","info","xpack"],"pid":1,"message":"Imported license information from Elasticsearch: mode: trial | status: active | expiry date: 2016-12-23T00:26:08+00:00"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:21Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from green to yellow - Waiting for Monitoring Health Check","prevState":"green","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:21Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:21Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:21Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:24Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"Waiting for Monitoring Health Check"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["status","plugin:security@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["security","warning"],"pid":1,"message":"Generating a random key for xpack.security.encryptionKey. To prevent sessions from being invalidated on restart, please set xpack.security.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["security","warning"],"pid":1,"message":"Session cookies will be transmitted over insecure connections. This is not recommended."}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["status","plugin:console@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["status","plugin:timelion@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["listening","info"],"pid":1,"message":"Server running at http://0.0.0.0:5601"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:26:28Z","tags":["status","ui settings","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | [2016-11-23T00:26:29,811][INFO ][o.e.c.m.MetaDataMappingService] [NggTRJz] [.monitoring-es-2-2016.11.23/-AEPo8pmRoW0eDALvM_45g] update_mapping [index_stats]
elasticsearch_1  | [2016-11-23T00:26:29,918][INFO ][o.e.c.m.MetaDataMappingService] [NggTRJz] [.monitoring-es-2-2016.11.23/-AEPo8pmRoW0eDALvM_45g] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-23T00:26:29,981][INFO ][o.e.c.m.MetaDataMappingService] [NggTRJz] [.monitoring-es-2-2016.11.23/-AEPo8pmRoW0eDALvM_45g] update_mapping [indices_stats]
elasticsearch_1  | [2016-11-23T00:26:31,265][INFO ][o.e.c.m.MetaDataCreateIndexService] [NggTRJz] [.monitoring-kibana-2-2016.11.23] creating index, cause [auto(bulk api)], templates [.monitoring-kibana-2], shards [1]/[1], mappings [_default_, kibana_stats]

```

**Comment by deviantony** [comment]
After the max_vm_count step, please restart the stack using `docker-compose down && docker-compose up`.

**Comment by rohan04** [comment]
Yes , I tried those steps and I got the above error.

**Comment by deviantony** [comment]
Try to start logstash with a debug log level:

Update the logstash definition in the file `docker-compose.yml` with:

```yml
  logstash:
    build: logstash/
    command: -f /etc/logstash/conf.d/ --log.level debug
    volumes:
      - ./logstash/config:/etc/logstash/conf.d
    ports:
      - "5000:5000"
    networks:
      - docker_elk
    depends_on:
      - elasticsearch
```

And  restart the stack using `docker-compose down && docker-compose up`.

Note, I'm not sure if it's related but it seems there is an issue when elasticsearch is trying to boot:

```
elasticsearch_1  | org.elasticsearch.xpack.monitoring.exporter.ExportException: exporters are either not ready or faulty
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.export(Exporters.java:188) ~[x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.AgentService$ExportingWorker.run(AgentService.java:208) [x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
```

**Comment by rohan04** [comment]
Got the same error: 

```
Creating network "dockerelk_docker_elk" with driver "bridge"
Creating dockerelk_elasticsearch_1
Creating dockerelk_logstash_1
Creating dockerelk_kibana_1
Attaching to dockerelk_elasticsearch_1, dockerelk_kibana_1, dockerelk_logstash_1
kibana_1         | Stalling for Elasticsearch
elasticsearch_1  | [2016-11-23T00:50:22,785][INFO ][o.e.n.Node               ] [] initializing ...
elasticsearch_1  | [2016-11-23T00:50:23,051][INFO ][o.e.e.NodeEnvironment    ] [-ZgcK4i] using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/disk/by-uuid/21813bed-571e-4fcc-905d-32bc13641493)]], net usable_space [12.6gb], net total_space [29.4gb], spins? [possibly], types [ext4]
elasticsearch_1  | [2016-11-23T00:50:23,054][INFO ][o.e.e.NodeEnvironment    ] [-ZgcK4i] heap size [1007.3mb], compressed ordinary object pointers [true]
elasticsearch_1  | [2016-11-23T00:50:23,061][INFO ][o.e.n.Node               ] [-ZgcK4i] node name [-ZgcK4i] derived from node ID; set [node.name] to override
elasticsearch_1  | [2016-11-23T00:50:23,065][INFO ][o.e.n.Node               ] [-ZgcK4i] version[5.0.0], pid[1], build[253032b/2016-10-26T05:11:34.737Z], OS[Linux/3.13.0-46-generic/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/1.8.0_111/25.111-b14]
elasticsearch_1  | [2016-11-23T00:50:29,081][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [aggs-matrix-stats]
elasticsearch_1  | [2016-11-23T00:50:29,084][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [ingest-common]
elasticsearch_1  | [2016-11-23T00:50:29,089][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [lang-expression]
elasticsearch_1  | [2016-11-23T00:50:29,090][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [lang-groovy]
elasticsearch_1  | [2016-11-23T00:50:29,094][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [lang-mustache]
elasticsearch_1  | [2016-11-23T00:50:29,094][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [lang-painless]
elasticsearch_1  | [2016-11-23T00:50:29,096][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [percolator]
elasticsearch_1  | [2016-11-23T00:50:29,098][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [reindex]
elasticsearch_1  | [2016-11-23T00:50:29,098][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [transport-netty3]
elasticsearch_1  | [2016-11-23T00:50:29,099][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded module [transport-netty4]
elasticsearch_1  | [2016-11-23T00:50:29,110][INFO ][o.e.p.PluginsService     ] [-ZgcK4i] loaded plugin [x-pack]
elasticsearch_1  | [2016-11-23T00:50:30,683][WARN ][o.e.d.s.g.GroovyScriptEngineService] [groovy] scripts are deprecated, use [painless] scripts instead
elasticsearch_1  | [2016-11-23T00:50:39,188][INFO ][o.e.n.Node               ] [-ZgcK4i] initialized
elasticsearch_1  | [2016-11-23T00:50:39,191][INFO ][o.e.n.Node               ] [-ZgcK4i] starting ...
elasticsearch_1  | [2016-11-23T00:50:39,969][INFO ][o.e.t.TransportService   ] [-ZgcK4i] publish_address {172.18.0.2:9300}, bound_addresses {[::]:9300}
elasticsearch_1  | [2016-11-23T00:50:39,998][INFO ][o.e.b.BootstrapCheck     ] [-ZgcK4i] bound or publishing to a non-loopback or non-link-local address, enforcing bootstrap checks
elasticsearch_1  | [2016-11-23T00:50:43,716][INFO ][o.e.c.s.ClusterService   ] [-ZgcK4i] new_master {-ZgcK4i}{-ZgcK4iPSumdPzQ7bhqmiw}{hqR8sCA0SsuhCmYo-SEjnw}{172.18.0.2}{172.18.0.2:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
elasticsearch_1  | [2016-11-23T00:50:43,908][INFO ][o.e.h.HttpServer         ] [-ZgcK4i] publish_address {172.18.0.2:9200}, bound_addresses {[::]:9200}
elasticsearch_1  | [2016-11-23T00:50:43,936][INFO ][o.e.n.Node               ] [-ZgcK4i] started
kibana_1         | Starting Kibana
elasticsearch_1  | [2016-11-23T00:50:44,481][INFO ][o.e.g.GatewayService     ] [-ZgcK4i] recovered [0] indices into cluster_state
elasticsearch_1  | [2016-11-23T00:50:47,527][INFO ][o.e.m.j.JvmGcMonitorService] [-ZgcK4i] [gc][8] overhead, spent [265ms] collecting in the last [1s]
elasticsearch_1  | [2016-11-23T00:50:49,683][INFO ][o.e.x.m.e.Exporters      ] [-ZgcK4i] skipping exporter [default_local] as it isn't ready yet
elasticsearch_1  | [2016-11-23T00:50:49,684][ERROR][o.e.x.m.AgentService     ] [-ZgcK4i] exception when exporting documents
elasticsearch_1  | org.elasticsearch.xpack.monitoring.exporter.ExportException: exporters are either not ready or faulty
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.exporter.Exporters.export(Exporters.java:188) ~[x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at org.elasticsearch.xpack.monitoring.AgentService$ExportingWorker.run(AgentService.java:208) [x-pack-5.0.0.jar:5.0.0]
elasticsearch_1  | 	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:kibana@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["reporting","warning"],"pid":1,"message":"Generating a random key for xpack.reporting.encryptionKey. To prevent pending reports from failing on restart, please set xpack.reporting.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:50:55Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from uninitialized to yellow - Waiting for Elasticsearch","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | [2016-11-23T00:50:59,970][INFO ][o.e.c.m.MetaDataCreateIndexService] [-ZgcK4i] [.monitoring-es-2-2016.11.23] creating index, cause [auto(bulk api)], templates [.monitoring-es-2], shards [1]/[1], mappings [_default_, node, shards, index_stats, index_recovery, cluster_state, cluster_stats, indices_stats, node_stats]
elasticsearch_1  | [2016-11-23T00:51:00,257][INFO ][o.e.c.m.MetaDataCreateIndexService] [-ZgcK4i] [.monitoring-data-2] creating index, cause [auto(bulk api)], templates [.monitoring-data-2], shards [1]/[1], mappings [node, cluster_info, kibana]
elasticsearch_1  | [2016-11-23T00:51:00,602][INFO ][o.e.c.m.MetaDataMappingService] [-ZgcK4i] [.monitoring-es-2-2016.11.23/nafAKXpPRwGrxcZt-o274w] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-23T00:51:00,778][INFO ][o.e.c.m.MetaDataMappingService] [-ZgcK4i] [.monitoring-es-2-2016.11.23/nafAKXpPRwGrxcZt-o274w] update_mapping [node_stats]
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:02Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:02Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:02Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:02Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from yellow to yellow - No existing Kibana index found","prevState":"yellow","prevMsg":"Waiting for Elasticsearch"}
elasticsearch_1  | [2016-11-23T00:51:02,067][WARN ][o.e.d.i.m.StringFieldMapper$TypeParser] The [string] field is deprecated, please use [text] or [keyword] instead on [buildNum]
elasticsearch_1  | [2016-11-23T00:51:02,070][INFO ][o.e.c.m.MetaDataCreateIndexService] [-ZgcK4i] [.kibana] creating index, cause [api], templates [], shards [1]/[1], mappings [server, config]
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:02Z","tags":["status","plugin:elasticsearch@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Kibana index ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:03Z","tags":["license","info","xpack"],"pid":1,"message":"Imported license information from Elasticsearch: mode: trial | status: active | expiry date: 2016-12-23T00:50:50+00:00"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:03Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"yellow","message":"Status changed from green to yellow - Waiting for Monitoring Health Check","prevState":"green","prevMsg":"Ready"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:03Z","tags":["status","plugin:xpack_main@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:03Z","tags":["status","plugin:graph@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:03Z","tags":["status","plugin:reporting@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"No existing Kibana index found"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:05Z","tags":["status","plugin:monitoring@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from yellow to green - Ready","prevState":"yellow","prevMsg":"Waiting for Monitoring Health Check"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:08Z","tags":["status","plugin:security@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:08Z","tags":["security","warning"],"pid":1,"message":"Generating a random key for xpack.security.encryptionKey. To prevent sessions from being invalidated on restart, please set xpack.security.encryptionKey in kibana.yml"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:08Z","tags":["security","warning"],"pid":1,"message":"Session cookies will be transmitted over insecure connections. This is not recommended."}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:08Z","tags":["status","plugin:console@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:09Z","tags":["status","plugin:timelion@5.0.0","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:09Z","tags":["listening","info"],"pid":1,"message":"Server running at http://0.0.0.0:5601"}
kibana_1         | {"type":"log","@timestamp":"2016-11-23T00:51:09Z","tags":["status","ui settings","info"],"pid":1,"state":"green","message":"Status changed from uninitialized to green - Ready","prevState":"uninitialized","prevMsg":"uninitialized"}
elasticsearch_1  | [2016-11-23T00:51:11,015][INFO ][o.e.c.m.MetaDataMappingService] [-ZgcK4i] [.monitoring-es-2-2016.11.23/nafAKXpPRwGrxcZt-o274w] update_mapping [index_stats]
elasticsearch_1  | [2016-11-23T00:51:11,081][INFO ][o.e.c.m.MetaDataMappingService] [-ZgcK4i] [.monitoring-es-2-2016.11.23/nafAKXpPRwGrxcZt-o274w] update_mapping [cluster_stats]
elasticsearch_1  | [2016-11-23T00:51:11,157][INFO ][o.e.c.m.MetaDataMappingService] [-ZgcK4i] [.monitoring-es-2-2016.11.23/nafAKXpPRwGrxcZt-o274w] update_mapping [indices_stats]
elasticsearch_1  | [2016-11-23T00:51:13,113][INFO ][o.e.c.m.MetaDataCreateIndexService] [-ZgcK4i] [.monitoring-kibana-2-2016.11.23] creating index, cause [auto(bulk api)], templates [.monitoring-kibana-2], shards [1]/[1], mappings [_default_, kibana_stats]
logstash_1       | Sending Logstash logs to /var/log/logstash which is now configured via log4j2.properties.
dockerelk_logstash_1 exited with code 1
```

**Comment by rohan04** [comment]
From the error stack you posted earlier, I'm guessing its looking for Java 1.8, my java_home is set to 1.7. Does Java_version have anything to do with this.

at java.lang.Thread.run(Thread.java:745) [?:1.8.0_111]

**Comment by deviantony** [comment]
No it's the java version shipped in the containers. No idea here... you'll have to find a way to access logstash logs to find out the issue.

**Comment by rohan04** [comment]
Ok thanks for looking into this , I'll keep digging further.

**Comment by antoineco** [comment]
@rohan04: Logstash 5.2 seems to require the [`log4j2.properties`](https://github.com/elastic/logstash-docker/tree/master/build/logstash/config) file in its config directory (see README).

If you mount the entire `logstash/config/` you have to make sure you include this file.

Can you try that on the current `master` and close this issue if it solves your problem?

**Comment by charan-code** [comment]
kibana_1         | {"type":"error","@timestamp":"2020-05-19T11:07:00Z","tags":["fatal"],"pid":1,"level":"fatal","error":{"message":"child \"elasticsearch\" fails because [\"hosts\" is not allowed]","name":"ValidationError","stack":"ValidationError: child \"elasticsearch\" fails because [\"hosts\" is not allowed]\n    at Object.exports.process (/usr/share/kibana/node_modules/joi/lib/errors.js:181:19)\n    at _validateWithOptions (/usr/share/kibana/node_modules/joi/lib/any.js:651:31)\n    at root.validate (/usr/share/kibana/node_modules/joi/lib/index.js:121:23)\n    at Config._commit (/usr/share/kibana/src/server/config/config.js:119:35)\n    at Config.set (/usr/share/kibana/src/server/config/config.js:89:10)\n    at Config.extendSchema (/usr/share/kibana/src/server/config/config.js:62:10)\n    at /usr/share/kibana/src/plugin_discovery/plugin_config/extend_config_service.js:22:12\n    at next (native)\n    at step (/usr/share/kibana/src/plugin_discovery/plugin_config/extend_config_service.js:45:191)\n    at /usr/share/kibana/src/plugin_discovery/plugin_config/extend_config_service.js:45:361"},"message":"child \"elasticsearch\" fails because [\"hosts\" is not allowed]"}
dockerelk_elasticsearch_1 exited with code 1
dockerelk_kibana_1 exited with code 1

**Comment by antoineco** [comment]
@charan-code this issue is closed, please open a new issue.

**Comment by charan-code** [comment]
Thanks alotttt sir, will try and tell u
